code/modules/projectiles/gun_helpers.dm

```diff
@@ -167,8 +167,9 @@ As sniper rifles have both and weapon mods can change them as well. ..() deals w
 		var/obj/item/attachable/attachment = attachments[slot]
 		if (!attachment || !attachment.light_mod)
 			continue
-		bearer.SetLuminosity(0, FALSE, src)
-		SetLuminosity(attachment.light_mod)
+
+		attachment.activate_attachment(src, bearer)
```

Comment:
could use `turn_off = TRUE` to short circuit turning it off if it's actually off

Reply:
Does this work without casting the attachment to the flashlight one? (not that that would be bad per se 🤷 )

Reply:
Yeahhh not all of them have it, scratch that